src/compile/scale.ts

```diff
@@ -191,79 +292,114 @@ export function type(specifiedType: ScaleType, fieldDef: FieldDef, channel: Chan
     if (supportScaleType(channel, specifiedType)) {
       return specifiedType;
     } else {
-      const newScaleType = defaultType(fieldDef, channel, mark, canHaveBandSize);
+      const newScaleType = defaultProperty.type(fieldDef, channel, mark, canHaveBandSize);
       log.warn(log.message.scaleTypeNotWorkWithChannel(channel, specifiedType, newScaleType));
       return newScaleType;
     }
   }
 
-  return defaultType(fieldDef, channel, mark, canHaveBandSize);
+  return defaultProperty.type(fieldDef, channel, mark, canHaveBandSize);
 }
 
-function defaultType(fieldDef: FieldDef, channel: Channel, mark: Mark, canHaveBandSize: boolean): ScaleType {
-  if (contains([ROW, COLUMN], channel)) {
-    return ScaleType.BAND;
-  }
-  if (channel === SHAPE) {
-    return ScaleType.ORDINAL_LOOKUP;
+namespace defaultProperty {
+  /**
+   * Determine appropriate default scale type.
+   */
+  export function type(fieldDef: FieldDef, channel: Channel, mark: Mark, canHaveBandSize: boolean): ScaleType {
+    if (contains([ROW, COLUMN], channel)) {
+      return ScaleType.BAND;
+    }
+    if (channel === SHAPE) {
+      return ScaleType.ORDINAL_LOOKUP;
+    }
+
+    switch (fieldDef.type) {
+      case NOMINAL:
+        if (channel === COLOR) {
+          return ScaleType.ORDINAL_LOOKUP;
+        }
+        return ordinalToContinuousType(channel, mark, canHaveBandSize);
+      case ORDINAL:
+        if (channel === COLOR) {
+          // TODO: check if this is still true
+          return ScaleType.LINEAR; // ordinal has order, so use interpolated ordinal color scale.
+        }
+        return ordinalToContinuousType(channel, mark, canHaveBandSize);
+      case TEMPORAL:
+        if (channel === COLOR) {
+          // FIXME: or sequential?
+          return ScaleType.TIME; // time has order, so use interpolated color scale.
+        }
+
+        switch (fieldDef.timeUnit) {
+          // These time unit use discrete scale by default
+          case TimeUnit.HOURS:
+          case TimeUnit.DAY:
+          case TimeUnit.MONTH:
+          case TimeUnit.QUARTER:
+            return ordinalToContinuousType(channel, mark, canHaveBandSize);
+        }
+        return ScaleType.TIME;
+
+      case QUANTITATIVE:
+        return ScaleType.LINEAR;
+    }
+
+    // should never reach this
+    return null;
```

Comment:
Not sure whether you can sue never here. https://basarat.gitbooks.io/typescript/content/docs/types/never.html